### PR TITLE
Update web samples to use ASP.NET 6 minimal host

### LINKED ISSUE #5837: Update ASP.NET Core samples to new minimal hosting model
ASP.NET Core 6 comes with a new, simpler hosting model API (see https://docs.microsoft.com/en-us/aspnet/core/migration/50-to-60?view=aspnetcore-6.0&tabs=visual-studio#new-hosting-model). We can update our samples to show how to use the NServiceBus generic host integration with the new hosting API and also make the samples easier to read.

### PR DESCRIPTION
**Pull request by timbussmann**
Fixes #5837 

Updates web samples to use the ASP.NET 6 minimal host API.

Also adds a dedicated documentation page for ASP.NET apps to show different configuration options and how to make use of the `IMessageSession` from DI.

### CODE REVIEW

File: samples/web/asp-web-application/Core_7/WebApp/Program.cs [hunk 1]
```diff
@@ -0,0 +1,24 @@
+﻿using Microsoft.AspNetCore.Builder;
```
Comment (timbussmann) [inline samples/web/asp-web-application/Core_7/WebApp/Program.cs:1]: thinking a bit more about this, maybe we should leave the v7 version on the "old" generic host model to have both being shown.

Comment (timbussmann) [reply samples/web/asp-web-application/Core_7/WebApp/Program.cs:1]: discussed this and we agreed to keep the samples on the minimal version across the board

File: nservicebus/hosting/asp-net.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+related:
+ - nservicebus/hosting/web-application
+ - samples/web
+---
+
+NServiceBus can be easily integrated with [ASP.NET Core](https://asp.net) applications.
+
+## Minimal Host
+
+ASP.NET 6 introduces a new hosting API with the [ASP.NET minimal APIs](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/minimal-apis?view=aspnetcore-6.0). The [NServiceBus.Extensions.Hosting package](/nservicebus/hosting/extensions-hosting.md) is fully compatible with the minimal API's `WebApplication` host:
```
Comment (kbaley) [inline nservicebus/hosting/asp-net.md:15]:
```suggestion
ASP.NET 6 introduced a new hosting API with the [ASP.NET minimal APIs](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/minimal-apis?view=aspnetcore-6.0). The [NServiceBus.Extensions.Hosting package](/nservicebus/hosting/extensions-hosting.md) is fully compatible with the minimal API's `WebApplication` host:
```

File: nservicebus/hosting/web-application.md [hunk 3]
```diff
@@ -12,31 +12,33 @@ related:
...
 
 * Any service which is registered with the NServiceBus dependency injection can be injected into NServiceBus message handlers but not the web request handlers.
 * Any service which is registered with the web host dependency injection mechanism can be injected into web request handlers but not the NServiceBus message handlers.
 * Any service which is registered with both dependency injection instances can be resolved in both web request handlers and NServiceBus message bus handlers _but_:
   * these will be different object instances with different lifetimes.
-  * even if the services are registered with a singleton lifetime, there will still be one created for each dependency injection instance.
+  * even if the services are registered with a singleton lifetime, there will still be one created for each dependency injection instance unless the instance is explicitly managed by the caller.
   * if a service must be shared and a single instance, it must be created externally during the web application host startup, and that specific instance must be registered in both dependency injection instances.
+
+## Endpoint Lifecycle
```
Comment (kbaley) [inline nservicebus/hosting/web-application.md:42]:
```suggestion
## Endpoint lifecycle
```

File: samples/web/asp-mvc-application/Core_8/Server/Server.csproj [hunk 4]
```diff
@@ -1,7 +1,7 @@
-<Project Sdk="Microsoft.NET.Sdk">
+﻿<Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
     <OutputType>Exe</OutputType>
-    <TargetFrameworks>net6.0;netcoreapp3.1</TargetFrameworks>
+	<TargetFramework>net6.0</TargetFramework>
```
Comment (kbaley) [inline samples/web/asp-mvc-application/Core_8/Server/Server.csproj:4]:
```suggestion
    <TargetFramework>net6.0</TargetFramework>
```

File: samples/web/asp-web-application/Core_7/Shared/Shared.csproj [hunk 5]
```diff
@@ -1,7 +1,7 @@
-<Project Sdk="Microsoft.NET.Sdk">
+﻿<Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
-    <TargetFrameworks>net6.0;netcoreapp3.1</TargetFrameworks>
-    <LangVersion>7.3</LangVersion>
+    <TargetFramework>net6.0</TargetFramework>
+	<LangVersion>7.3</LangVersion>
```
Comment (kbaley) [inline samples/web/asp-web-application/Core_7/Shared/Shared.csproj:4]:
```suggestion
    <LangVersion>7.3</LangVersion>
```

File: samples/web/asp-web-application/Core_7/WebApp/WebApp.csproj [hunk 6]
```diff
@@ -1,7 +1,7 @@
 <Project Sdk="Microsoft.NET.Sdk.Web">
   <PropertyGroup>
-    <TargetFrameworks>net6.0;netcoreapp3.1</TargetFrameworks>
-    <LangVersion>7.3</LangVersion>
+    <TargetFramework>net6.0</TargetFramework>
+	<LangVersion>7.3</LangVersion>
```
Comment (kbaley) [inline samples/web/asp-web-application/Core_7/WebApp/WebApp.csproj:4]:
```suggestion
    <LangVersion>7.3</LangVersion>
```

File: samples/web/asp-web-application/sample.md [hunk 7]
```diff
@@ -1,37 +1,41 @@
...
+When running the solution a new browser window/tab opens, as well as a console application.
 
 Enter the number "1" into the text box in the browser and click "Go". Notice the result "None" appears, as shown:
 
 ![AsyncPages sample running](async-pages-running.png "AsyncPages sample running")
 
 Changing the number in the text box from even to odd numbers changes the result in the Server console.
 
-The web page renders synchronously; from the user's perspective, the interaction is synchronous and blocking, even though behind the scenes NServiceBus is doing asynchronous messaging.
+The web page renders synchronously; from the user's perspective, the interaction is synchronous and blocking, even though behind the scenes NServiceBus is doing asynchronous send-reply pattern using the [callbacks package](/nservicebus/messaging/callbacks.md).
```
Comment (kbaley) [inline samples/web/asp-web-application/sample.md:24]:
```suggestion
The web page renders synchronously; from the user's perspective, the interaction is synchronous and blocking, even though behind the scenes NServiceBus is implementing an asynchronous send-reply pattern using the [callbacks package](/nservicebus/messaging/callbacks.md).
```

File: samples/web/send-from-aspnetcore-webapi/Core_7/Endpoint/Endpoint.csproj [hunk 8]
```diff
@@ -1,8 +1,8 @@
 <Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
-    <TargetFrameworks>net6.0;netcoreapp3.1</TargetFrameworks>
+	<TargetFramework>net6.0</TargetFramework>
+	<LangVersion>7.3</LangVersion>
```
Comment (kbaley) [inline samples/web/send-from-aspnetcore-webapi/Core_7/Endpoint/Endpoint.csproj:4]:
```suggestion
    <TargetFramework>net6.0</TargetFramework>
    <LangVersion>7.3</LangVersion>
```

File: samples/web/send-from-aspnetcore-webapi/Core_7/Shared/Shared.csproj [hunk 9]
```diff
@@ -1,7 +1,7 @@
 <Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
-    <TargetFrameworks>net6.0;netcoreapp3.1</TargetFrameworks>
-    <LangVersion>7.3</LangVersion>
+	<TargetFramework>net6.0</TargetFramework>
+	<LangVersion>7.3</LangVersion>
```
Comment (kbaley) [inline samples/web/send-from-aspnetcore-webapi/Core_7/Shared/Shared.csproj:4]:
```suggestion
    <TargetFramework>net6.0</TargetFramework>
    <LangVersion>7.3</LangVersion>
```

File: samples/web/send-from-aspnetcore-webapi/Core_7/WebApp/WebApp.csproj [hunk 10]
```diff
@@ -1,7 +1,7 @@
-<Project Sdk="Microsoft.NET.Sdk.Web">
+﻿<Project Sdk="Microsoft.NET.Sdk.Web">
   <PropertyGroup>
-    <TargetFrameworks>net6.0;netcoreapp3.1</TargetFrameworks>
-    <LangVersion>7.3</LangVersion>
+    <TargetFramework>net6.0</TargetFramework>
+	<LangVersion>7.3</LangVersion>
```
Comment (kbaley) [inline samples/web/send-from-aspnetcore-webapi/Core_7/WebApp/WebApp.csproj:4]:
```suggestion
    <TargetFramework>net6.0</TargetFramework>
    <LangVersion>7.3</LangVersion>
```

File: samples/web/send-from-aspnetcore-webapi/Core_8/Shared/Shared.csproj [hunk 11]
```diff
@@ -1,7 +1,7 @@
 <Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
-    <TargetFrameworks>net6.0;netcoreapp3.1</TargetFrameworks>
-    <LangVersion>7.3</LangVersion>
+    <TargetFramework>net6.0</TargetFramework>
+	<LangVersion>7.3</LangVersion>
```
Comment (kbaley) [inline samples/web/send-from-aspnetcore-webapi/Core_8/Shared/Shared.csproj:4]:
```suggestion
    <TargetFramework>net6.0</TargetFramework>
    <LangVersion>7.3</LangVersion>
```

**Review by kbaley — APPROVED** [review]

**Review by cquirosj — APPROVED** [review]

**Pull request merged**